pkg/server/conn.go

```diff
@@ -349,9 +350,13 @@ func (cc *clientConn) Close() error {
 	return closeConn(cc, connections)
 }
 
-// closeConn should be idempotent.
+// closeConn is idempotent and thread-safe.
```

Comment:
Do we need to ensure that concurrent invocations of `closeConn` are blocked until the first `closeConn` is finished? like `sync.Once`. I'm not sure if the second caller see `closeConn` is finished it will access other objects and will race with the running `closeConn`

Reply:
Good point! I'll update the code.